cmd/solution/name.go

```diff
@@ -49,7 +51,7 @@ func getSolutionNameFromArgs(cmd *cobra.Command, args []string, flagName string)
 		}
 		// We only want to append .dev for subscribing/unsubscribing commands
 		if solutionTag != "" && solutionTag != "stable" && (commandName == "subscribe" || commandName == "unsubscribe") {
-			if solutionTag == "dev" {
+			if (solutionTag == "dev") || (config.GetCurrentContext().EnvType == "dev") {
```

Comment:
we shouldn't take this branch if `(config.GetCurrentContext().EnvType == "dev")` -- for the `dev` environment, the `.dev` suffix is not supported at all. If the envType is `dev`, we should keep the solution name as is (just `name`).

Reply:
In this case should the checking the entype be done on line 51?

Reply:
Yes, if you are referring to the old line 51 or new 53, then yes, you can add `config.GetCurrentContext().EnvType != "dev"` as another and-ed condition; I would recommend to make it the first one (since all of the rest applies only to non-dev environments). Note that we want != "dev" (so that both explicit "prod" and implicit "prod"/default will still follow the default).
Also, it may be a good idea to define the environment types as constants